src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1473,27 +1498,36 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
                                   TimeoutExchangeSec(),
                                   policy_manager_->RetrySequenceDelaysSeconds(),
                                   application_manager_);
-#else   // PROPRIETARY_MODE
-  LOG4CXX_ERROR(logger_, "HTTP policy");
-  EndpointUrls urls;
-  policy_manager_->GetUpdateUrls("0x07", urls);
 
-  if (urls.empty()) {
-    LOG4CXX_ERROR(logger_, "Service URLs are empty! NOT sending PT to mobile!");
+  OnUpdateRequestSentToMobile();
+}
+
+void PolicyHandler::OnNextRetry() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const std::string url = ChooseUrl();
+  ApplicationSharedPtr app = ChooseApplication();
+  uint32_t app_id = app->app_id();
```

Comment:
@LitvinenkoIra 
1. You use app before checking that it is not null 
2. app_id can be const